### ISSUE TITLE
AWS CLI credentials other than $ HOME or less

### ISSUE THREAD

**Issue by ggtakec** [s3fs-fuse/s3fs-fuse#1860]
There is a request to read the credential information(and region information) from the files(.aws/credential and config), which are the configure files of AWS CLI.
Those are different from the format of the passwd-s3fs file, but with the same credential information.

I would like to consider whether or not this should be supported.

- Related - #1855

**Comment by ggtakec** [comment]
Duplicate of #1122.

**Comment by siketyan** [comment]
@ggtakec 
You s3fs-fuse already supports .aws/credentials file to read credentials from `$HOME` e.g. `/home/$USER/.aws/credentials`.
The option I suggested is to read this file from other location.

Background:
In MSYS2 environment, getting the home directory returns "/home/$USER" instead of "C:\Users\$USER" for their compatibility. This behaviour blocks s3fs-fuse on Windows to read the file out of MSYS2 shell.

**Comment by ggtakec** [comment]
Reopen this.

**Comment by gaul** [comment]
We should follow the precedent of AWS CLI (and related SDKs) as closely as possible:

https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html#cli-configure-files-where

As I understand it `get_access_keys` should consider `%UserProfile%` when using Windows.  In addition to consistency, this will allow the project to transition to the AWS C++ SDK some day #1068.

**Comment by ggtakec** [comment]
For Windows, let's use `%UserProfile%` instead of `$HOME`.(After #1855 pr)
